static/src/js/library/search/Search.vue

```diff
@@ -0,0 +1,225 @@
+<template>
+  <section>
+    <div class="container">
+      <div class="search-guide">
+        <h3>Search Guide</h3>
+        <div class="row">
+          <div class="col-md-6">
+            <kbd>+</kbd> signifies AND operation<br>
+            <kbd>|</kbd> signifies OR operation<br>
+            <kbd>-</kbd> negates a single token<br>
+            <kbd>"</kbd> wraps a number of tokens to signify a phrase for searching<br>
+          </div>
+          <div class="col-md-6">
+            <kbd>*</kbd> at the end of a term signifies a prefix query<br>
+            <kbd>(</kbd> and <kbd>)</kbd> signify precedence<br>
+            <kbd>~<var>num</var></kbd> after a word signifies edit distance (fuzziness)<br>
+            <kbd>~<var>num</var></kbd> after a phrase signifies slop amount<br>
+          </div>
+        </div>
+      </div>
+      <form class="search-form" v-on:submit.prevent="handleSearch(0)">
+        <div class="form-group">
+          <input type="text" class="form-control" placeholder="Search..." :value="searchQuery" @input="handleSearchQueryChange">
+          <input
+            type="radio"
+            id="kind-form"
+            name="form"
+            :value="searchType"
+            @input="handleTypeChange"
+            :checked="searchType=='form'"
+          >
+          <label for="kind-form">Form</label>
+          <input
+            type="radio"
+            id="kind-lemma"
+            name="lemma"
+            :value="searchType"
+            @input="handleTypeChange"
+            :checked="searchType=='lemma'"
+          >
+          <label for="kind-lemma">Lemma (Greek only)</label>
+        </div>
+      </form>
+      <text-loader v-if="loading" size="7px" margin="1px" />
+      <div v-if="showSearchResults" class="row">
+        <div v-if="!results.length && !textGroups.length" class="col-sm-12 text-center">
+          <p class="no-results">No results found. Please try again.</p>
+        </div>
+          <search-text-groups
+            :textGroups=textGroups
+            :handleSearch=handleSearch
+            :showClear=showClear
+            :showTextGroups=showTextGroups
+            :handleshowTextGroupsChange=handleshowTextGroupsChange
+          />
+        <div v-if="results.length" class="col-md-9">
+          <search-pagination
+            :startIndex=startIndex
+            :endIndex=endIndex
+            :totalResults=totalResults
+            :pageNum=pageNum
+            :totalPages=totalPages
+            :hasNext=hasNext
+            :hasPrev=hasPrev
+            :handleSearch=handleSearch
+          />
+          <div>
+            <text-loader v-if="secondLoading" size="7px" margin="1px" />
+            <div class="result" v-if="!secondLoading" v-for="result in results" :key="result.link">
+              <div class="passage-heading">
+                <h2>
+                  <a :href="''+result.link+'?q='+searchQuery+'&amp;qk='+searchType+''">
```

Comment:
Could we move the link generation out of the template and to a helper method on the component (e.g., `passageLink(result.link)`)?

Reply:
Done. https://github.com/scaife-viewer/scaife-viewer/pull/342/commits/ba86f4334167e23b345c4250da77e7778128a884